File: app/src/main/java/com/alphadevelopmentsolutions/frcscout/Fragments/ScoutCardAutoFragment.java
package com.alphadevelopmentsolutions.frcscout.Fragments;

import android.content.Context;
import android.net.Uri;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import com.alphadevelopmentsolutions.frcscout.Classes.ScoutCard;
import com.alphadevelopmentsolutions.frcscout.R;
import com.google.gson.Gson;

/**
 * A simple {@link Fragment} subclass.
 * Activities that contain this fragment must implement the
 * {@link ScoutCardAutoFragment.OnFragmentInteractionListener} interface
 * to handle interaction events.
 * Use the {@link ScoutCardAutoFragment#newInstance} factory method to
 * create an instance of this fragment.
 */
public class ScoutCardAutoFragment extends MasterFragment {
    // TODO: Rename parameter arguments, choose names that match
    // the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
    private static final String ARG_PARAM1 = "param1";

    // TODO: Rename and change types of parameters
    private String scoutCardJson;

    private OnFragmentInteractionListener mListener;

    public ScoutCardAutoFragment() {
        // Required empty public constructor
    }

    /**
     * Use this factory method to create a new instance of
     * this fragment using the provided parameters.
     *
     * @param scoutCardJson scout card json
     * @return A new instance of fragment ScoutCardAutoFragment.
     */
    public static ScoutCardAutoFragment newInstance(String scoutCardJson) {
        ScoutCardAutoFragment fragment = new ScoutCardAutoFragment();
        Bundle args = new Bundle();
        args.putString(ARG_PARAM1, scoutCardJson);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            scoutCardJson = getArguments().getString(ARG_PARAM1);
        }

        //load the passed scout card
        if(scoutCardJson != null && !scoutCardJson.equals(""))
            scoutCard = new Gson().fromJson(scoutCardJson, ScoutCard.class);
    }

    private ScoutCard scoutCard;

    //Exit Habitat
    private TextView autonomousExitHabitatTextView;

    private Button autonomousExitHabitatNoButton;
    private Button autonomousExitHabitatYesButton;

    //Hatch Panels
    private TextView autonomousHatchPanelsPickupTextView;
    private TextView autonomousHatchPanelsSecuredAttemptsTextView;
    private TextView autonomousHatchPanelsSecuredTextView;

    private Button autonomousHatchPanelsPickupMinusOneButton;
    private Button autonomousHatchPanelsPickupPlusOneButton;

    private Button autonomousHatchPanelsSecuredMinusOneButton;
    private Button autonomousHatchPanelsSecuredPlusOneButton;

    private Button autonomousHatchPanelsSecuredAttemptsMinusOneButton;
    private Button autonomousHatchPanelsSecuredAttemptsPlusOneButton;

    //Cargo
    private TextView autonomousCargoPickupTextView;
    private TextView autonomousCargoStoredAttemptsTextView;
    private TextView autonomousCargoStoredTextView;


    private Button autonomousCargoPickupMinusOneButton;
    private Button autonomousCargoPickupPlusOneButton;

    private Button autonomousCargoStoredAttemptsMinusOneButton;
    private Button autonomousCargoStoredAttemptsPlusOneButton;

    private Button autonomousCargoStoredMinusOneButton;
    private Button autonomousCargoStoredPlusOneButton;



    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view = inflater.inflate(R.layout.fragment_scout_card_auto, container, false);

        //Exit Habitat
        autonomousExitHabitatTextView = view.findViewById(R.id.AutonomousExitHabitatTextView);

        autonomousExitHabitatNoButton = view.findViewById(R.id.AutonomousExitHabitatNoButton);
        autonomousExitHabitatYesButton = view.findViewById(R.id.AutonomousExitHabitatYesButton);

        autonomousExitHabitatNoButton.setOnClickListener(new View.OnClickListener()
        {
            @Override
            public void onClick(View v)
            {
                autonomousExitHabitatTextView.setText(R.string.no);
            }
        });

        autonomousExitHabitatYesButton.setOnClickListener(new View.OnClickListener()
        {
            @Override
            public void onClick(View v)
            {
                autonomousExitHabitatTextView.setText(R.string.yes);
            }
        });


        //Hatch Panels
        autonomousHatchPanelsPickupTextView = view.findViewById(R.id.AutonomousHatchPanelsPickupTextView);
        autonomousHatchPanelsSecuredAttemptsTextView = view.findViewById(R.id.AutonomousHatchPanelsSecuredAttemptsTextView);
        autonomousHatchPanelsSecuredTextView = view.findViewById(R.id.AutonomousHatchPanelsSecuredTextView);


        autonomousHatchPanelsPickupMinusOneButton = view.findViewById(R.id.AutonomousHatchPanelsPickupMinusOneButton);
        autonomousHatchPanelsPickupPlusOneButton = view.findViewById(R.id.AutonomousHatchPanelsPickupPlusOneButton);

        autonomousHatchPanelsSecuredAttemptsMinusOneButton = view.findViewById(R.id.AutonomousHatchPanelsSecuredAttemptsMinusOneButton);
        autonomousHatchPanelsSecuredAttemptsPlusOneButton = view.findViewById(R.id.AutonomousHatchPanelsSecuredAttemptsPlusOneButton);

        autonomousHatchPanelsSecuredMinusOneButton = view.findViewById(R.id.AutonomousHatchPanelsSecuredMinusOneButton);
        autonomousHatchPanelsSecuredPlusOneButton = view.findViewById(R.id.AutonomousHatchPanelsSecuredPlusOneButton);

        autonomousHatchPanelsPickupMinusOneButton.setOnClickListener(new View.OnClickListener()
        {
            @Override
            public void onClick(View v)
            {
                autonomousHatchPanelsPickupTextView.setText((Integer.parseInt(autonomousHatchPanelsPickupTextView.getText().toString()) - 1) + "");
            }
        });

        autonomousHatchPanelsPickupPlusOneButton.setOnClickListener(new View.OnClickListener()
        {
            @Override
            public void onClick(View v)
            {
                autonomousHatchPanelsPickupTextView.setText((Integer.parseInt(autonomousHatchPanelsPickupTextView.getText().toString()) + 1) + "");
            }
        });

        autonomousHatchPanelsSecuredAttemptsMinusOneButton.setOnClickListener(new View.OnClickListener()
        {
            @Override
            public void onClick(View v)
            {
                autonomousHatchPanelsSecuredAttemptsTextView.setText((Integer.parseInt(autonomousHatchPanelsSecuredAttemptsTextView.getText().toString()) - 1) + "");
            }
        });

        autonomousHatchPanelsSecuredAttemptsPlusOneButton.setOnClickListener(new View.OnClickListener()
        {
            @Override
            public void onClick(View v)
            {
                autonomousHatchPanelsSecuredAttemptsTextView.setText((Integer.parseInt(autonomousHatchPanelsSecuredAttemptsTextView.getText().toString()) + 1) + "");
            }
        });

        autonomousHatchPanelsSecuredMinusOneButton.setOnClickListener(new View.OnClickListener()
        {
            @Override
            public void onClick(View v)
            {
                autonomousHatchPanelsSecuredTextView.setText((Integer.parseInt(autonomousHatchPanelsSecuredTextView.getText().toString()) - 1) + "");
            }
        });

        autonomousHatchPanelsSecuredPlusOneButton.setOnClickListener(new View.OnClickListener()
        {
            @Override
            public void onClick(View v)
            {
                autonomousHatchPanelsSecuredTextView.setText((Integer.parseInt(autonomousHatchPanelsSecuredTextView.getText().toString()) + 1) + "");
            }
        });


        //Cargo
        autonomousCargoPickupTextView = view.findViewById(R.id.AutonomousCargoPickupTextView);
        autonomousCargoStoredAttemptsTextView = view.findViewById(R.id.AutonomousCargoStoredAttemptsTextView);
        autonomousCargoStoredTextView = view.findViewById(R.id.AutonomousCargoStoredTextView);


        autonomousCargoPickupMinusOneButton = view.findViewById(R.id.AutonomousCargoPickupMinusOneButton);
        autonomousCargoPickupPlusOneButton = view.findViewById(R.id.AutonomousCargoPickupPlusOneButton);

        autonomousCargoStoredAttemptsPlusOneButton = view.findViewById(R.id.AutonomousCargoStoredAttemptsPlusOneButton);
        autonomousCargoStoredAttemptsMinusOneButton = view.findViewById(R.id.AutonomousCargoStoredAttemptsMinusOneButton);

        autonomousCargoStoredMinusOneButton = view.findViewById(R.id.AutonomousCargoStoredMinusOneButton);
        autonomousCargoStoredPlusOneButton = view.findViewById(R.id.AutonomousCargoStoredPlusOneButton);

        autonomousCargoPickupMinusOneButton.setOnClickListener(new View.OnClickListener()
        {
            @Override
            public void onClick(View v)
            {
                autonomousCargoPickupTextView.setText((Integer.parseInt(autonomousCargoPickupTextView.getText().toString()) - 1) + "");
            }
        });

        autonomousCargoPickupPlusOneButton.setOnClickListener(new View.OnClickListener()
        {
            @Override
            public void onClick(View v)
            {
                autonomousCargoPickupTextView.setText((Integer.parseInt(autonomousCargoPickupTextView.getText().toString()) + 1) + "");
            }
        });

        autonomousCargoStoredAttemptsMinusOneButton.setOnClickListener(new View.OnClickListener()
        {
            @Override
            public void onClick(View v)
            {
                autonomousCargoStoredAttemptsTextView.setText((Integer.parseInt(autonomousCargoStoredAttemptsTextView.getText().toString()) - 1) + "");
            }
        });

        autonomousCargoStoredAttemptsPlusOneButton.setOnClickListener(new View.OnClickListener()
        {
            @Override
            public void onClick(View v)
            {
                autonomousCargoStoredAttemptsTextView.setText((Integer.parseInt(autonomousCargoStoredAttemptsTextView.getText().toString()) + 1) + "");
            }
        });

        autonomousCargoStoredMinusOneButton.setOnClickListener(new View.OnClickListener()
        {
            @Override
            public void onClick(View v)
            {
                autonomousCargoStoredTextView.setText((Integer.parseInt(autonomousCargoStoredTextView.getText().toString()) - 1) + "");
            }
        });

        autonomousCargoStoredPlusOneButton.setOnClickListener(new View.OnClickListener()
        {
            @Override
            public void onClick(View v)
            {
                autonomousCargoStoredTextView.setText((Integer.parseInt(autonomousCargoStoredTextView.getText().toString()) + 1) + "");
            }
        });

        return view;
    }

    //region Getters

    public boolean getAutonomousExitHab()
    {
        return autonomousExitHabitatTextView.getText().toString().equals(context.getResources().getString(R.string.yes));
    }

    public int getAutonomousHatchPanelsSecured()
    {
        return Integer.parseInt(autonomousHatchPanelsSecuredTextView.getText().toString());
    }

    public int getAutonomousHatchPanelsDropped()
    {
        return Integer.parseInt(autonomousHatchPanelsSecuredAttemptsTextView.getText().toString());
    }

    public int getAutonomousHatchPanelsPickedUp()
    {
        return Integer.parseInt(autonomousHatchPanelsPickupTextView.getText().toString());
    }

    public int getAutonomousCargoStored()
    {
        return Integer.parseInt(autonomousCargoStoredTextView.getText().toString());
    }

    public int getAutonomousCargoDropped()
    {
        return Integer.parseInt(autonomousCargoStoredAttemptsTextView.getText().toString());
    }

    public int getAutonomousCargoPickedUp()
    {
        return Integer.parseInt(autonomousCargoPickupTextView.getText().toString());
    }

    //endregion

    /**
     * Validates all fields on the form for saving
     * @return boolean if fields valid
     */
    public boolean validateFields()
    {
        //placeholder in case of future validation needed
        return true;
    }

    // TODO: Rename method, update argument and hook method into UI event
    public void onButtonPressed(Uri uri) {
        if (mListener != null) {
            mListener.onFragmentInteraction(uri);
        }
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        if (context instanceof OnFragmentInteractionListener) {
            mListener = (OnFragmentInteractionListener) context;
        } else {
            throw new RuntimeException(context.toString()
                    + " must implement OnFragmentInteractionListener");
        }
    }

    @Override
    public void onDetach() {
        super.onDetach();
        mListener = null;
    }

    /**
     * This interface must be implemented by activities that contain this
     * fragment to allow an interaction in this fragment to be communicated
     * to the activity and potentially other fragments contained in that
     * activity.
     * <p>
     * See the Android Training lesson <a href=
     * "http://developer.android.com/training/basics/fragments/communicating.html"
     * >Communicating with Other Fragments</a> for more information.
     */
    public interface OnFragmentInteractionListener {
        // TODO: Update argument type and name
        void onFragmentInteraction(Uri uri);
    }
}

File: app/src/main/java/com/alphadevelopmentsolutions/frcscout/Fragments/ScoutCardAutoFragment.kt
package com.alphadevelopmentsolutions.frcscout.Fragments

import android.content.Context
import android.net.Uri
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView

import com.alphadevelopmentsolutions.frcscout.Classes.Tables.ScoutCard
import com.alphadevelopmentsolutions.frcscout.R

/**
 * A simple [Fragment] subclass.
 * Activities that contain this fragment must implement the
 * [ScoutCardAutoFragment.OnFragmentInteractionListener] interface
 * to handle interaction events.
 * Use the [ScoutCardAutoFragment.newInstance] factory method to
 * create an instance of this fragment.
 */
class ScoutCardAutoFragment : MasterFragment()
{

    private var mListener: OnFragmentInteractionListener? = null

    //Exit Habitat
    private var autonomousExitHabitatTextView: TextView? = null

    private var autonomousExitHabitatNoButton: Button? = null
    private var autonomousExitHabitatYesButton: Button? = null

    //Hatch Panels
    private var autonomousHatchPanelsPickupTextView: TextView? = null
    private var autonomousHatchPanelsSecuredAttemptsTextView: TextView? = null
    private var autonomousHatchPanelsSecuredTextView: TextView? = null

    private var autonomousHatchPanelsPickupMinusOneButton: Button? = null
    private var autonomousHatchPanelsPickupPlusOneButton: Button? = null

    private var autonomousHatchPanelsSecuredMinusOneButton: Button? = null
    private var autonomousHatchPanelsSecuredPlusOneButton: Button? = null

    private var autonomousHatchPanelsSecuredAttemptsMinusOneButton: Button? = null
    private var autonomousHatchPanelsSecuredAttemptsPlusOneButton: Button? = null

    //Cargo
    private var autonomousCargoPickupTextView: TextView? = null
    private var autonomousCargoStoredAttemptsTextView: TextView? = null
    private var autonomousCargoStoredTextView: TextView? = null


    private var autonomousCargoPickupMinusOneButton: Button? = null
    private var autonomousCargoPickupPlusOneButton: Button? = null

    private var autonomousCargoStoredAttemptsMinusOneButton: Button? = null
    private var autonomousCargoStoredAttemptsPlusOneButton: Button? = null

    private var autonomousCargoStoredMinusOneButton: Button? = null
    private var autonomousCargoStoredPlusOneButton: Button? = null

    //region Getters

    val autonomousExitHab: Boolean
        get() = autonomousExitHabitatTextView!!.text.toString() == context.getString(R.string.yes)

    val autonomousHatchPanelsSecured: Int
        get() = Integer.parseInt(autonomousHatchPanelsSecuredTextView!!.text.toString())

    val autonomousHatchPanelsDropped: Int
        get() = Integer.parseInt(autonomousHatchPanelsSecuredAttemptsTextView!!.text.toString())

    val autonomousHatchPanelsPickedUp: Int
        get() = Integer.parseInt(autonomousHatchPanelsPickupTextView!!.text.toString())

    val autonomousCargoStored: Int
        get() = Integer.parseInt(autonomousCargoStoredTextView!!.text.toString())

    val autonomousCargoDropped: Int
        get() = Integer.parseInt(autonomousCargoStoredAttemptsTextView!!.text.toString())

    val autonomousCargoPickedUp: Int
        get() = Integer.parseInt(autonomousCargoPickupTextView!!.text.toString())

    override fun onCreate(savedInstanceState: Bundle?)
    {
        super.onCreate(savedInstanceState)
    }


    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View?
    {
        // Inflate the layout for this fragment
        val view = inflater.inflate(R.layout.fragment_scout_card_auto, container, false)

        joinLoadingThread()

        //Exit Habitat
        autonomousExitHabitatTextView = view.findViewById(R.id.AutonomousExitHabitatTextView)

        autonomousExitHabitatNoButton = view.findViewById(R.id.AutonomousExitHabitatNoButton)
        autonomousExitHabitatYesButton = view.findViewById(R.id.AutonomousExitHabitatYesButton)

        autonomousExitHabitatNoButton!!.setOnClickListener {
            if (scoutCard == null || scoutCard!!.isDraft)
                autonomousExitHabitatTextView!!.setText(R.string.no)
        }

        autonomousExitHabitatYesButton!!.setOnClickListener {
            if (scoutCard == null || scoutCard!!.isDraft)
                autonomousExitHabitatTextView!!.setText(R.string.yes)
        }


        //Hatch Panels
        autonomousHatchPanelsPickupTextView = view.findViewById(R.id.AutonomousHatchPanelsPickupTextView)
        autonomousHatchPanelsSecuredAttemptsTextView = view.findViewById(R.id.AutonomousHatchPanelsSecuredAttemptsTextView)
        autonomousHatchPanelsSecuredTextView = view.findViewById(R.id.AutonomousHatchPanelsSecuredTextView)


        autonomousHatchPanelsPickupMinusOneButton = view.findViewById(R.id.AutonomousHatchPanelsPickupMinusOneButton)
        autonomousHatchPanelsPickupPlusOneButton = view.findViewById(R.id.AutonomousHatchPanelsPickupPlusOneButton)

        autonomousHatchPanelsSecuredAttemptsMinusOneButton = view.findViewById(R.id.AutonomousHatchPanelsSecuredAttemptsMinusOneButton)
        autonomousHatchPanelsSecuredAttemptsPlusOneButton = view.findViewById(R.id.AutonomousHatchPanelsSecuredAttemptsPlusOneButton)

        autonomousHatchPanelsSecuredMinusOneButton = view.findViewById(R.id.AutonomousHatchPanelsSecuredMinusOneButton)
        autonomousHatchPanelsSecuredPlusOneButton = view.findViewById(R.id.AutonomousHatchPanelsSecuredPlusOneButton)

        autonomousHatchPanelsPickupMinusOneButton!!.setOnClickListener {
            if (scoutCard == null || scoutCard!!.isDraft)
                autonomousHatchPanelsPickupTextView!!.text = (Integer.parseInt(autonomousHatchPanelsPickupTextView!!.text.toString()) - 1).toString() + ""
        }

        autonomousHatchPanelsPickupPlusOneButton!!.setOnClickListener {
            if (scoutCard == null || scoutCard!!.isDraft)
                autonomousHatchPanelsPickupTextView!!.text = (Integer.parseInt(autonomousHatchPanelsPickupTextView!!.text.toString()) + 1).toString() + ""
        }

        autonomousHatchPanelsSecuredAttemptsMinusOneButton!!.setOnClickListener {
            if (scoutCard == null || scoutCard!!.isDraft)
                autonomousHatchPanelsSecuredAttemptsTextView!!.text = (Integer.parseInt(autonomousHatchPanelsSecuredAttemptsTextView!!.text.toString()) - 1).toString() + ""
        }

        autonomousHatchPanelsSecuredAttemptsPlusOneButton!!.setOnClickListener {
            if (scoutCard == null || scoutCard!!.isDraft)
                autonomousHatchPanelsSecuredAttemptsTextView!!.text = (Integer.parseInt(autonomousHatchPanelsSecuredAttemptsTextView!!.text.toString()) + 1).toString() + ""
        }

        autonomousHatchPanelsSecuredMinusOneButton!!.setOnClickListener {
            if (scoutCard == null || scoutCard!!.isDraft)
                autonomousHatchPanelsSecuredTextView!!.text = (Integer.parseInt(autonomousHatchPanelsSecuredTextView!!.text.toString()) - 1).toString() + ""
        }

        autonomousHatchPanelsSecuredPlusOneButton!!.setOnClickListener {
            if (scoutCard == null || scoutCard!!.isDraft)
                autonomousHatchPanelsSecuredTextView!!.text = (Integer.parseInt(autonomousHatchPanelsSecuredTextView!!.text.toString()) + 1).toString() + ""
        }


        //Cargo
        autonomousCargoPickupTextView = view.findViewById(R.id.AutonomousCargoPickupTextView)
        autonomousCargoStoredAttemptsTextView = view.findViewById(R.id.AutonomousCargoStoredAttemptsTextView)
        autonomousCargoStoredTextView = view.findViewById(R.id.AutonomousCargoStoredTextView)


        autonomousCargoPickupMinusOneButton = view.findViewById(R.id.AutonomousCargoPickupMinusOneButton)
        autonomousCargoPickupPlusOneButton = view.findViewById(R.id.AutonomousCargoPickupPlusOneButton)

        autonomousCargoStoredAttemptsPlusOneButton = view.findViewById(R.id.AutonomousCargoStoredAttemptsPlusOneButton)
        autonomousCargoStoredAttemptsMinusOneButton = view.findViewById(R.id.AutonomousCargoStoredAttemptsMinusOneButton)

        autonomousCargoStoredMinusOneButton = view.findViewById(R.id.AutonomousCargoStoredMinusOneButton)
        autonomousCargoStoredPlusOneButton = view.findViewById(R.id.AutonomousCargoStoredPlusOneButton)

        autonomousCargoPickupMinusOneButton!!.setOnClickListener {
            if (scoutCard == null || scoutCard!!.isDraft)
                autonomousCargoPickupTextView!!.text = (Integer.parseInt(autonomousCargoPickupTextView!!.text.toString()) - 1).toString() + ""
        }

        autonomousCargoPickupPlusOneButton!!.setOnClickListener {
            if (scoutCard == null || scoutCard!!.isDraft)
                autonomousCargoPickupTextView!!.text = (Integer.parseInt(autonomousCargoPickupTextView!!.text.toString()) + 1).toString() + ""
        }

        autonomousCargoStoredAttemptsMinusOneButton!!.setOnClickListener {
            if (scoutCard == null || scoutCard!!.isDraft)
                autonomousCargoStoredAttemptsTextView!!.text = (Integer.parseInt(autonomousCargoStoredAttemptsTextView!!.text.toString()) - 1).toString() + ""
        }

        autonomousCargoStoredAttemptsPlusOneButton!!.setOnClickListener {
            if (scoutCard == null || scoutCard!!.isDraft)
                autonomousCargoStoredAttemptsTextView!!.text = (Integer.parseInt(autonomousCargoStoredAttemptsTextView!!.text.toString()) + 1).toString() + ""
        }

        autonomousCargoStoredMinusOneButton!!.setOnClickListener {
            if (scoutCard == null || scoutCard!!.isDraft)
                autonomousCargoStoredTextView!!.text = (Integer.parseInt(autonomousCargoStoredTextView!!.text.toString()) - 1).toString() + ""
        }

        autonomousCargoStoredPlusOneButton!!.setOnClickListener {
            if (scoutCard == null || scoutCard!!.isDraft)
                autonomousCargoStoredTextView!!.text = (Integer.parseInt(autonomousCargoStoredTextView!!.text.toString()) + 1).toString() + ""
        }

        //scoutcard loaded, populate fields
        if (scoutCard != null)
        {
            autonomousHatchPanelsPickupTextView!!.text = scoutCard!!.autonomousHatchPanelsPickedUp.toString()
            autonomousHatchPanelsSecuredAttemptsTextView!!.text = scoutCard!!.autonomousHatchPanelsSecuredAttempts.toString()
            autonomousHatchPanelsSecuredTextView!!.text = scoutCard!!.autonomousHatchPanelsSecured.toString()

            autonomousCargoPickupTextView!!.text = scoutCard!!.autonomousCargoPickedUp.toString()
            autonomousCargoStoredAttemptsTextView!!.text = scoutCard!!.autonomousCargoStoredAttempts.toString()
            autonomousCargoStoredTextView!!.text = scoutCard!!.autonomousCargoStored.toString()

            autonomousExitHabitatTextView!!.setText(if (scoutCard!!.autonomousExitHabitat) R.string.yes else R.string.no)
        }

        return view
    }

    //endregion

    /**
     * Validates all fields on the form for saving
     * @return boolean if fields valid
     */
    fun validateFields(): Boolean
    {
        //placeholder in case of future validation needed
        return true
    }

    // TODO: Rename method, update argument and hook method into UI event
    fun onButtonPressed(uri: Uri)
    {
        if (mListener != null)
        {
            mListener!!.onFragmentInteraction(uri)
        }
    }

    override fun onAttach(context: Context?)
    {
        super.onAttach(context)
        if (context is OnFragmentInteractionListener)
        {
            mListener = context
        } else
        {
            throw RuntimeException(context!!.toString() + " must implement OnFragmentInteractionListener")
        }
    }

    override fun onDetach()
    {
        super.onDetach()
        mListener = null
    }

    /**
     * This interface must be implemented by activities that contain this
     * fragment to allow an interaction in this fragment to be communicated
     * to the activity and potentially other fragments contained in that
     * activity.
     *
     *
     * See the Android Training lesson [Communicating with Other Fragments](http://developer.android.com/training/basics/fragments/communicating.html) for more information.
     */
    interface OnFragmentInteractionListener
    {
        // TODO: Update argument type and name
        fun onFragmentInteraction(uri: Uri)
    }

    companion object
    {

        /**
         * Use this factory method to create a new instance of
         * this fragment using the provided parameters.
         *
         * @param scoutCard
         * @return A new instance of fragment ScoutCardAutoFragment.
         */
        fun newInstance(scoutCard: ScoutCard?): ScoutCardAutoFragment
        {
            val fragment = ScoutCardAutoFragment()
            val args = Bundle()
            args.putString(MasterFragment.ARG_PARAM_SCOUT_CARD_JSON, MasterFragment.toJson(scoutCard))
            fragment.arguments = args
            return fragment
        }
    }
}// Required empty public constructor
